operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java

```diff
@@ -74,18 +74,28 @@ private static Step scaleDownIfNecessary(
     List<ServerShutdownInfo> serversToStop = getServersToStop(info, factory.shutdownInfos);
 
     if (!serversToStop.isEmpty()) {
-      insert(steps,
-          Step.chain(createProgressingStartedEventStep(info, MANAGED_SERVERS_STARTING_PROGRESS_REASON, true,
-          null), new ServerDownIteratorStep(factory.shutdownInfos, null)));
+      if (isServiceOnlyServers(serversToStop)) {
+        steps.add(new ServerDownIteratorStep(factory.shutdownInfos, null));
+      } else {
+        insert(steps,
+                Step.chain(createProgressingStartedEventStep(info, MANAGED_SERVERS_STARTING_PROGRESS_REASON, true,
+                        null), new ServerDownIteratorStep(factory.shutdownInfos, null)));
+      }
     }
 
     return Step.chain(steps.toArray(new Step[0]));
   }
 
+  private static boolean isServiceOnlyServers(List<ServerShutdownInfo> serversToStop) {
```

Comment:
i think it would make the code more easy to read if this method is renamed to `isAllServersServiceOnly`

Reply:
The logic makes sense to me, but I agree that the name needs to be longer and/or have a comment that describes the intention. I think this is looking for the condition where for all of the servers in the stop list we are only interested in deleting the service rather than deleting a pod. Is that right?

Reply:
> The logic makes sense to me, but I agree that the name needs to be longer and/or have a comment that describes the intention. I think this is looking for the condition where for all of the servers in the stop list we are only interested in deleting the service rather than deleting a pod. Is that right?

Yes. Additionally if precreateServerService option is set to `true` and service doesn't exist (for the first time), ManagedServerDown step creates the service with `isPreserveService` option set to `true` (instead of deleting the service and the pod). Please see below code snippet in `ServerDownIteratorStep.java`. Since I was only trying to address the issue of domain processing events being generated for every domain presence recheck, I decided not to change this logic. Please let me know if this needs to be changed.

```
  private StepAndPacket createManagedServerDownDetails(Packet packet, DomainPresenceInfo.ServerShutdownInfo ssi) {
    if (ssi.isServiceOnly()) {
      return new StepAndPacket(createServiceStep(ssi),
              createPacketForServer(packet, ssi));
    } else {
      return new StepAndPacket(new ServerDownStep(ssi.getName(), null), packet.clone());
    }
  }
```

Reply:
> i think it would make the code more easy to read if this method is renamed to `isAllServersServiceOnly`

I have refactored the code and removed the `isServiceOnlyServers` method in c26deeb. Please let me know if it's not clear.